discovery-provider/src/utils/session_manager.py

```diff
@@ -1,29 +1,51 @@
 from contextlib import contextmanager
+import logging  # pylint: disable=C0302
 from sqlalchemy import create_engine
+from sqlalchemy.event import listen
 from sqlalchemy.orm import sessionmaker
+from src.queries.search_config import set_search_similarity
+
+logger = logging.getLogger(__name__)
 
 
 class SessionManager:
     def __init__(self, db_url, db_engine_args):
         self._engine = create_engine(db_url, **db_engine_args)
         self._session_factory = sessionmaker(bind=self._engine)
+        listen(self._engine, 'connect', self.on_connect)
```

Comment:
just leaving here for reference
https://docs.sqlalchemy.org/en/13/core/events.html#sqlalchemy.events.PoolEvents.connect